solution.cpp

```diff
@@ -10,125 +10,212 @@
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
+using std::ifstream; 
+using std::ofstream; 
+
+
+
+int main()
+{
+	int num[] = {1};
+	int t,z;
+	ifstream myfile("Text.txt");
+	
+	// below counting code received from Matt
+    if (myfile.is_open())
+    {
+        
+        t = 0;
+        while (!myfile.eof()){
+            
+            myfile >> z;
+            t++;
+        }
+        cout<< t<<endl;
     }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+    myfile.close();
+    
+	ifstream inFile;
+	inFile.open("Text.txt");
+	for (int i = 0; i <= t; i++)
+	{
+		inFile >> num[i];
+	}
+	
+	inFile.close();
+
+	int myPrime[] = {0};
+	//int composites[] = {1};
+	int s = 0;
+	//int v = 0;
+
+	for (int j = 0; j < t; j++)
+	{
+		int number = num[j];
+		int w = j;
+		myPrime[w];
+		//composites[j];
+		int divider = 2;
+		//cout << j << ". Make a modus divider variable called divider and start it as " << divider << endl << endl;
+		if (number > 2)
+		{
+			for (int i = 2; i < number; i++)
+			{
+				int prime = number;
+				//cout << "Make " << prime << " into a variable called prime " << endl << endl;
+				prime = prime % divider;
+				//cout << number << "%" << divider << "= " << prime << endl << endl;
+				divider = divider++;
+				//cout << "Now increase divider by 1 into " << divider << endl << endl;
+				//cout << divider << endl << endl;
+
+
+				if (prime == 0)
+				{
+					//composites[j - v] = number;
+					divider = divider - 1;
+					/*cout << "However " << number << " can be divided by " << divider << "." << endl << endl;
+					cout << "In fact " << number << " / " << divider << " = " << number / divider << "." << endl << endl;
+					cout << "Therefore " << number << " is not a prime number." << endl << endl;*/
+					s = s++;
+					i = number;
+					
+					
+				}
+
+				else if (divider == number)
+				{
+					/*cout << i << ". " << number << " cannot be divided by " << "2" << " to " << divider - 1 << endl << endl;
+					cout << number << " is a prime number." << endl;*/
+					myPrime[w - s] = number;
+					//v = v++;
+				}
+
+
+			}
+		}
+		else if (number > 1)
+		{
+			myPrime[w - s] = number;
+			//v = v++;
+			//cout << number << " is a prime number" << endl;
+
+		}
+		else 
+		{
+			s = s++;
+			//composites[j - v] = number;
+			//cout << number << " is not prime number" << endl;
+		}
+
+		
+	}
+		ofstream outFile;
+		outFile.open("prime.txt");
+		for (int j = 0; j < t -s; j++)
+		{
+			outFile << myPrime[j] << endl;
+		}
+		outFile.close();
```

Comment:
also, you weren't supposed to do all this in `main()` and it probably would have been *much* simpler if you had split it out into functions.  it does look  like you thought it through pretty thoroughly though, so, i appreciate that :-)